src/t8_geometry/t8_geometry_implementations/t8_geometry_examples.cxx

```diff
@@ -0,0 +1,127 @@
+/*
+  This file is part of t8code.
+  t8code is a C library to manage a collection (a forest) of multiple
+  connected adaptive space-trees of general element classes in parallel.
+
+  Copyright (C) 2023 the developers
+
+  t8code is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2 of the License, or
+  (at your option) any later version.
+
+  t8code is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with t8code; if not, write to the Free Software Foundation, Inc.,
+  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+#include <t8_geometry/t8_geometry_implementations/t8_geometry_examples.hxx>
+#include <t8_geometry/t8_geometry_helpers.h>
+#include <t8_vec.h>
+
+/**
+ * Map the faces of a unit square to a disk.
+ * \param [in]  cmesh      The cmesh in which the point lies.
+ * \param [in]  gtreeid    The global tree (of the cmesh) in which the reference point is.
+ * \param [in]  ref_coords  Array of \a dimension many entries, specifying a point in [0,1]^dimension.
+ * \param [out] out_coords  The mapped coordinates in physical space of \a ref_coords.
+ */
+void
+t8_geometry_squared_disk::t8_geom_evaluate (t8_cmesh_t cmesh,
+                                            t8_gloidx_t gtreeid,
+                                            const double *ref_coords,
+                                            double out_coords[3]) const
+{
+  double              n[3];     /* Normal vector. */
+  double              r[3];     /* Radial vector. */
+  double              s[3];     /* Vector on the sphere. */
+  double              p[3];     /* Vector on the plane. */
+
+  t8_locidx_t         ltreeid = t8_cmesh_get_local_id (cmesh, gtreeid);
+  double             *tree_vertices =
+    t8_cmesh_get_tree_vertices (cmesh, ltreeid);
+
+  t8_geom_linear_interpolation (ref_coords, tree_vertices, 3, 2, p);
+
+  /* Center square. */
+  if (gtreeid == 0) {
+    out_coords[0] = p[0];
+    out_coords[1] = p[1];
+    out_coords[2] = 0.0;
+
+    return;
+  }
+
+  /* Four squares framing the central one. */
+  {
+    const double        center_ref[3] = { 0.5, 0.5, 0.0 };
+    t8_geom_linear_interpolation (center_ref, tree_vertices, 3, 2, n);
+
+    /* Normalize vector `n`. */
+    const double        norm = sqrt (n[0] * n[0] + n[1] * n[1]);
```

Comment:
If you zero-initialize n, you should be able to use t8_vec_norm

Reply:
Indeed. But this is a no-brainer and wouldn't shorten or simplify the code by calling the function.